lib/promise-hash.js

```diff
@@ -1,18 +1,13 @@
 'use strict';
 
-Promise.hash = function(iteratable) {
-  return new Promise((resolve, reject) => {
-    const result = {};
-    const promiseCount = Object.keys(iteratable).length;
-    let cursor = 0;
-
-    for (const promise in iteratable) {
-      iteratable[promise].then(promiseResult => {
-        result[promise] = promiseResult;
-        cursor++;
-
-        if (cursor >= promiseCount) return resolve(result);
-      }).catch(reject);
-    }
+Promise.hash = Promise.hash || function (hashOfPromises) {
```

Comment:
Good thinking here, I hadn't considered that there already may be a polyfill in place!